archunit/src/main/java/com/tngtech/archunit/core/domain/properties/CanBeAnnotated.java

```diff
@@ -19,12 +19,15 @@
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.util.Collection;
+import java.util.HashSet;
```

Comment:
Since there is no way to comment on commit messages in GitHub :disappointed: I'll just add this comment here:

I don't understand the list in the first commit message:

> This could have happened in the following setting:
> 1. A JavaClass is meta annotated with a cyclic annotation, e.g., `@Retention`.
> 2. The JavaClass, its annotation and meta annotation are imported.
> 3. The JavaClass is checked for a meta annotation that is not the above mentioned cyclic annotation.

Wouldn't point 1. be already perfectly fine as a reason? I don't understand what info 2. and 3. add, because in the end it's always if you call isMetaAnnotatedWith(..) and a recursively annotated annotation like `@Retention` is checked in the process, no?

Also there is a typo in the first commit message ;-) 'exectution'